### PR TITLE
Support facets for search (ported from older feature/facetSearch bran…

### PR DESCRIPTION
**Pull request by bbfsdev**
…ch).

### CODE REVIEW

File: consts/consts.go [hunk 1]
```diff
@@ -53,15 +53,18 @@ const (
...
-	SCT_TWEET     = "R_TWEET"
+	SCT_TWEET = "R_TWEET"
 
 	// Content Role types
 	CR_LECTURER = "LECTURER"
 
 	// Persons patterns
 	P_RAV = "rav"
 
+	// Persons UIDs
```
Comment (davgur) [inline consts/consts.go:64]: I think you can use [this](https://github.com/Bnei-Baruch/archive-backend/blob/9b798744d583894fda1a2439468cb2ea42c9dd5e/mdb/registry.go#L88)

**Review by davgur — APPROVED** [review]

File: search/engine.go [hunk 2]
```diff
@@ -632,9 +640,9 @@ func uniqueHitsByMdbUid(hits []*elastic.SearchHit, indexesToIgnore []string, typ
 			if err := json.Unmarshal(*hit.Source, &mdbUid); err == nil {
 				if mdbUid.MDB_UID != "" {
 					// Uncomment for debug
-					/*if _, ok := mdbMap[mdbUid.MDB_UID]; ok {
-						log.Infof("Found duplicate of %+v", hit)
-					}*/
+					if dupHit, ok := mdbMap[mdbUid.MDB_UID]; ok {
```
Comment (YuriHechter) [inline search/engine.go:643]: comment out

File: search/query.go [hunk 3]
```diff
@@ -156,9 +162,24 @@ func ParseQuery(q string) Query {
 			}
 		}
 	}
+	// FixContentTypeToResultTypeFitlers(filters)
 	return Query{Term: strings.Join(terms, " "), ExactTerms: exactTerms, Original: q, Filters: filters}
 }
 
+/*
```
Comment (YuriHechter) [inline search/query.go:169]: remove?

File: utils/general.go [hunk 4]
```diff
@@ -134,6 +134,15 @@ func Int64InSlice(i int64, s []int64) bool {
 	return false
 }
 
+func StringInSlice(str string, s []string) bool {
+	for i := range s {
```
Comment (YuriHechter) [inline utils/general.go:138]: Consider change the function body to:
return utils.Contains(utils.Is(s), str)

Or replace all StringInSlice references to use utils.Contains like we already do in other parts of the project when we check that some string exist is slice.

**Review by YuriHechter — CHANGES_REQUESTED** [review]

File: consts/consts.go (hunk 1 above)
Comment (bbfsdev) [reply consts/consts.go:64]: Done, so we don't have hard coded UIDs but the code is not that preaty....

mdb.PERSONS_REGISTRY.ByPattern[consts.P_RAV].UID
mdb.PERSONS_REGISTRY.ByPattern[consts.P_RABASH].UID

File: consts/consts.go [hunk 5]
```diff
@@ -135,6 +138,44 @@ const (
...
+	CT_PICNIC,
+	CT_SONGS,
+	CT_SPECIAL_LESSON,
+	CT_UNITY_DAY,
+	CT_VIDEO_PROGRAM,
+	CT_VIRTUAL_LESSONS,
+	CT_WOMEN_LESSONS,
+}
+
+var MADOR_CT_TYPES = [...]string{
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply consts/consts.go:162]: Done.

File: search/engine.go (hunk 2 above)
Comment (bbfsdev) [reply search/engine.go:643]: Done.

File: search/query.go (hunk 3 above)
Comment (bbfsdev) [reply search/query.go:169]: Done.

**Comment by bbfsdev** [comment]
I have done all the fixes (or commented on them).
I merged (--amend) the changes into the last commit so you will not see the history.
Please review the code.

**Pull request merged**